Demo/Demo/ViewModels/BaseViewModel.swift

```diff
@@ -93,6 +93,25 @@ class BaseViewModel: ObservableObject, PayPalWebCheckoutDelegate, CardDelegate {
         let cardRequest = CardRequest(orderID: orderID, card: card, sca: .scaAlways)
         cardClient.approveOrder(request: cardRequest)
     }
+    
+    func vaultCard(
+        card: Card,
+        customerID: String? = nil
+    ) async {
+        do {
+            guard let config = await getCoreConfig() else { return }
```

Comment:
Should we throw an error if we can't get a config?

Reply:
I was just following example in card approval portion right above, but we should throw error for both.

Reply:
I've moved all vault delegation to separate CardVaultViewModel in this PR. 
Do you want me to handle these here or on the next PR?
https://github.com/paypal/paypal-ios/pull/183

Reply:
I was thinking that "base" part of `baseViewModel` should be extracted out, inherited by viewModels for each features and have CoreConfig be published property but I don't think that belongs in this PR.

Reply:
Gotcha - if it's covered in another PR that's fine, maybe for now just add a `TODO` to make that a bit more clear.